shared/js/background/features/fire-button.js

```diff
@@ -0,0 +1,305 @@
+        if (closeTabs) {
+            // remove the rest of the open tabs
+            await browser.tabs.remove(removeTabIds)
+        }
+    }
+
+    async showBurnAnimation () {
+        await browser.tabs.create({
+            active: true,
+            url: getExtensionURL('/html/fire.html')
+        })
+    }
+
+    /**
+     * Get options to display in the burn modal
+     * @returns {Promise<import('@duckduckgo/privacy-dashboard/schema/__generated__/schema.types').FireButtonData>}
+     */
+    async getBurnOptions () {
+        const ONE_HOUR = 60 * 60 * 1000
+        const [currentTab, allTabs, allCookies] = await Promise.all([
+            getCurrentTab(),
+            browser.tabs.query({}),
+            browser.cookies.getAll({})
+        ])
+        const openTabs = allTabs.filter(t => !t.pinned).length
+        const cookies = allCookies.reduce((sites, curr) => {
+            sites.add(getDomain(curr.domain, tldtsOptions))
+            return sites
+        }, new Set()).size
+        const pinnedTabs = allTabs.filter(t => t.pinned)
+        // Apply defaults for history and tab clearing
+        const { closeTabs, clearHistory, selectedOption } = this.getDefaultSettings()
+
+        const defaultStats = {
+            openTabs: closeTabs ? openTabs : 0,
+            cookies,
+            pinnedTabs: closeTabs ? pinnedTabs.length : 0,
+            clearHistory
+        }
+
+        /** @type {import('@duckduckgo/privacy-dashboard/schema/__generated__/schema.types').FireOption[]} */
+        const options = []
+        const currentTabUrl = currentTab?.url || ''
+        // only show the current site option if this an origin we can clear
+        if (currentTabUrl.startsWith('http:') || currentTabUrl.startsWith('https:')) {
+            const origins = getOriginsForUrl(currentTabUrl)
+            const tabsMatchingOrigin = allTabs.filter(tabMatchesHostFilter(origins)).length
+            const pinnedMatchingOrigin = pinnedTabs.filter(tabMatchesHostFilter(origins)).length
+            options.push({
+                name: 'CurrentSite',
+                options: {
+                    origins
+                },
+                descriptionStats: {
+                    ...defaultStats,
+                    openTabs: closeTabs ? tabsMatchingOrigin : 0,
+                    pinnedTabs: closeTabs ? pinnedMatchingOrigin : 0,
+                    cookies: 1,
+                    duration: 'all',
+                    site: getDomain(origins[0], tldtsOptions) || ''
+                }
+            })
+        }
+
+        options.push({
+            name: 'LastHour',
+            options: {
+                since: Date.now() - ONE_HOUR
+            },
+            descriptionStats: {
+                ...defaultStats,
+                duration: 'hour'
+            }
+        })
+        options.push({
+            name: 'Last24Hour',
+            options: {
+                since: Date.now() - (24 * ONE_HOUR)
```

Comment:
Nit: Perhaps add ONE_DAY_MS, ONE_WEEK_MS etc constants at the top of the file too?

Reply:
These values are only used in a single function call, so I think keeping them in there makes sense, so everything is self-contained. I did move them all to the top of the function though, so we define all the durations up-front.